cg/store/crud/read.py

```diff
@@ -1736,12 +1738,14 @@ def get_all_pools_to_deliver(self) -> list[Pool]:
         )
         return records.all()
 
-    def get_orders(self, limit: int | None = None) -> list[Order]:
+    def get_orders(self, orders_request: OrdersRequest | None = None) -> list[Order]:
```

Comment:
This query gets orders by workflow and not just all orders. We have function name for `get_samples` or `get_sample` reserved to either get all or one sample, for instance.

```suggestion
    def get_orders_by_workflow(self, workflow: str = None, limit = None) -> list[Order]:
```

To make the fuction more generally usable should pass on workflow and limit instead of an `OrderRequest` here. In case we need to re-use this functionality elsewhere we don't require the `OrderRequest` class,

Reply:
Yeah, it is sort of up for debate but I think I agree with you on this

Reply:
I will rewrite, and maybe have a separate function which just unpacks the request

Reply:
@ChrOertlin I don't really agree with the naming convention, though. Or would it not be unclearer if it was called `get_orders_by_workflow` but you can get all orders if you don't provide the workflow?

Reply:
Also, might not be more flexible to pass all parameters separately here, if we establish a standard where we have a request object containing all data that can be filtered upon. I will stick to the convention for now, but ideally, I would probably want to leave this pattern since it becomes very hard to read when you can filter on many fields.